pkg/tbtc/node.go

```diff
@@ -342,4 +342,8 @@ func (n *node) joinDKGIfEligible(seed *big.Int, startBlockNumber uint64) {
 	} else {
 		logger.Infof("not eligible for DKG with seed [0x%x]", seed)
 	}
+}
+
+func (n *node) DkgExecutor()(*dkg.Executor) {
+	return n.dkgExecutor
```

Comment:
I do not think this is terrible but I don't like it either. DKG executor is something that is very internal to `tbtc` package and I do not think it should leak out of there. On the other hand, there is certainly a challenge of how to expose some diagnostic information without leaking internals from a package. I do not think the current DKG pre-params pool size will be the last one.

Instead of exporting the executor, I would implement a diagnostics API allowing any package to register some endpoint under its namespace.

Do not stick to name/API, it's just a rough idea:

```
scheduler := generator.StartScheduler()

initializeMetrics(ctx, clientConfig, netProvider, blockCounter)
diagnosticsEndpoint = diagnostics.Initialize(ctx, clientConfig, netProvider, signing)

tbtc.Initialize(..., diagnosticsEndpoint.ForNamespace("tbtc"))
```

```
type Diagnostics interface {
  ForNamespace(namespace string) DiagnosticsNamespace
}

type DiagnosticsNamespace interface {
  Register(key string, value interface{})
}
```

And then, we would have:
```
{
   "client_info":{
      "chain_address":"0x407190f04d838Ec47dad4C0e0D2a9c49d7737479",
      "network_id":"16Uiu2HAm4WrMWq9Gc7WsymsqgrwbaD1KeW2eH8eY63DTZHXbRsuE",
      "version":"v1.3.1-5900-g7600556b1,
      "revision": 7600556b1,
   },
  {
    "tbtc": {
       "preParamsPoolSize":12,
     }
  }
}
```

Reply:
Ok, I've changed the approach, pls take a look at [157fd0b](https://github.com/keep-network/keep-core/pull/3240/commits/157fd0b37fe2144fe929300403510c1165a3eb6a)